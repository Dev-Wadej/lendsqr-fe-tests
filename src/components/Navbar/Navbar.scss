nav {
    font-family: 'Work sans', sans-serif;
    display: flex;
    margin-left: 3vw;
    background: #FFFFFF;
    box-shadow: 3px 0px 20px rgba(0, 0, 0, 0.04);
    justify-content: space-between;
    position: fixed;
    z-index: 2000;
    @media only screen and (max-width: 1200px) {
        & {
            margin-left: .5rem;
            width: 100vw;
        }
    }
    @media only screen and (max-width: 900px) {
        & {
            margin-left: 0rem;
            padding-left: .2rem;
            width: 100vw;
        }
    }
    .nav-logo {
        margin-top: 1.5rem;
        width: 17vw;
        background-color: #fff;
        @media only screen and (max-width: 1200px) {
            & {
                svg {
                    width: 70%;
                }
            }
        }
        @media only screen and (max-width: 800px) {
            & {
                svg {
                    width: 65%;
                }
            }
        }
        @media only screen and (max-width: 500px) {
            & {
                svg {
                    width: 100%;
                }
            }
        }
    }
    .nav-elements {
        width: 80vw;
        display: flex;
        align-items: center;
        justify-content: space-between;
        padding-right: 3vw;
        div {
            span {
                background-color: #39CDCC;
                display: inline-block;
                align-self: center;
                border-top-right-radius: .4rem;
                border-bottom-right-radius: .4rem;
                position: relative;
                left: -.4rem;
                padding: .5rem 1rem;
            }
            display: flex;
            width: 25rem;
            margin-top: 1.5rem;
            margin-left: 5rem;
            @media only screen and (max-width: 1100px) {
                & {
                    width: 18rem;
                    margin-top: 1rem;
                }
            }
            @media only screen and (max-width: 900px) {
                & {
                    width: 14rem;
                    margin-left: 3rem;
                }
            }
            @media only screen and (max-width: 600px) {
                & {
                    margin-left: 0rem;
                    width: 15rem;
                    span {
                        display: none;
                    }
                }
            }
            input {
                width: 100%;
                padding: .5rem;
                border-radius: .4rem;
                border: 1px solid #dde2f0;
                @media only screen and (max-width: 900px) {
                    & {
                        padding: .4rem;
                    }
                }
                @media only screen and (max-width: 500px) {
                    & {
                        padding: .6rem;
                    }
                }
            }
            input::placeholder {
                color: #545F7D;
                opacity: 0.7;
                font-size: 90%;
                position: relative;
                left: .8rem;
                @media only screen and (max-width: 1100px) {
                    & {
                        font-size: 80%;
                    }
                }
                @media only screen and (max-width: 900px) {
                    & {
                        font-size: 70%;
                    }
                }
                @media only screen and (max-width: 800px) {
                    & {
                        font-size: 60%;
                    }
                }
            }
            input:focus {
                outline: none;
            }
        }
    }
}

.navbar-ul {
    display: flex;
    align-items: center;
    gap: 2rem;
    margin-top: 1.5rem;
    li {
        a {
            font-family: 'Roboto', sans-serif;
            border-bottom: 1px solid #213F7D;
            color: rgba(33, 63, 125, 1);
            padding-bottom: 1px;
        }
        img {
            display: inline-block;
            border-radius: 50%;
        }
        span {
            position: relative;
            top: -1rem;
        }
    }
    @media only screen and (max-width: 1100px) {
        & {
            gap: 1.6rem;
            a {
                font-size: .7rem;
            }
            li:nth-child(3) {
                span {
                    font-size: .7rem;
                }
                img {
                    display: inline-block;
                    width: 40%;
                    height: 40%;
                    margin-right: .2rem;
                }
            }
        }
    }
    @media only screen and (max-width: 900px) {
        & {
            gap: 1.3rem;
            a {
                font-size: .5rem;
            }
            li:nth-child(3) {
                span {
                    font-size: .5rem;
                    margin-top: .3rem;
                }
                img {
                    display: inline-block;
                    width: 30%;
                    height: 30%;
                    margin-right: .15rem;
                }
            }
        }
    }
    @media only screen and (max-width: 800px) {
        & {
            gap: .8rem;
            li {
                svg[width] {
                    width: 22;
                    height: 24;
                    fill: none;
                }
            }
        }
    }
    @media only screen and (max-width: 500px) {
        & {
            flex-direction: column-reverse;
            margin-top: 0;
            gap: 0rem;
            padding-top: .4rem;
        }
    }
}