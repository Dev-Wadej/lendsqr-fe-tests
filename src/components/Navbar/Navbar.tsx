import React from 'react'
import { Outlet } from 'react-router-dom'
import './Navbar.scss'
import {ReactComponent as CompanyLogo} from '../../data/svgs/companyLogo.svg'
import {ReactComponent as Search} from '../../data/svgs/search.svg'
import {ReactComponent as ArrowDown} from '../../data/svgs/arrowdown.svg'
import Human from '../../data/images/human.jpg'
import {ReactComponent as Bell} from '../../data/svgs/bell.svg'
import Sidebar from '../Sidebar/Sidebar'

const Navbar = () => {
  return (
      <>
      <nav>
        <div className='nav-logo'>
          <CompanyLogo/>
        </div>
        <div className='nav-elements'>
          <div>
            <input type="text" placeholder='Search For Anything' />
            <span>
            <Search/>
              </span>
          </div>
          <ul className='navbar-ul'>
            <li><a href="#">Docs</a></li>
            <li><Bell/></li>
            <li> 
              <img src={Human} alt="The User's thumbnail" /> 
               <span> Adedeji </span>
              <span><ArrowDown/></span>
            </li>
            
          </ul>

        </div>
      </nav>
      <Sidebar/>

      <Outlet/>
      </>

  )
}

export default Navbar
