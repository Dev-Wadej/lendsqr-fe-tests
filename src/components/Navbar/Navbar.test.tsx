import Navbar from './Navbar'
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';



afterEach(cleanup)
test('Form component snapshot HTML', () => {
  const output = renderer.create(<Navbar/>).toJSON();
  expect(output).toMatchSnapshot();
});