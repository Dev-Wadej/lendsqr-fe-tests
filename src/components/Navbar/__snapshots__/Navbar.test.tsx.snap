// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot HTML 1`] = `
<nav
  className="navbar-top"
>
  <div
    className="nav-logo"
  >
    <svg>
      companyLogo.svg
    </svg>
  </div>
  <div
    className="nav-elements"
  >
    <div>
      <input
        placeholder="Search For Anything"
        type="text"
      />
      <span>
        <svg>
          search.svg
        </svg>
      </span>
    </div>
    <ul
      className="navbar-ul"
    >
      <li>
        <a
          href="#"
        >
          Docs
        </a>
      </li>
      <li>
        <svg>
          bell.svg
        </svg>
      </li>
      <li>
        <img
          alt="The User's thumbnail"
          src="human.jpg"
        />
        <span>
           Adedeji 
        </span>
        <span>
          <svg>
            arrowdown.svg
          </svg>
        </span>
      </li>
    </ul>
  </div>
</nav>
`;
